services/seatool-sink/handlers/seaToolEvent.js

```diff
@@ -11,18 +11,19 @@ function myHandler(event) {
   }
   console.log("Received event:", JSON.stringify(event, null, 2));
   const topicName = event.topic.replace(topicPrefix, "");
-  const value = JSON.parse(event.value);
+  const value = event?.value ? JSON.parse(event.value) : event;
+  if (!event?.value) value.seaToolDelete = "true";
   console.log(`Type: ${topicName} Topic: ${event.topic}`);
 
-  const pk = value.STATE_PLAN.ID_NUMBER.toUpperCase();
-  const changedDate = value.STATE_PLAN.CHANGED_DATE;
+  const pk = JSON.parse(event.key).toUpperCase();
+  const changedDate = value?.STATE_PLAN?.CHANGED_DATE || value.timestamp;
```

Comment:
I do wonder what the downstream effects of using value.timestamp in lieu of STATE_PLAN.CHANGED_DATE would be as far as if we could get a real SEATOOL event (non delete) that is missing the STATE_PLAN.CHANGED_DATE and we should have bailed (thrown an error even) rather than using the event timestamp.

Thats probably not likely and would be the source data fault.....id just be worried about accidentally processing something without us realizing the data fault. But perhaps im reading this wrong too.